### PR TITLE
furnace: 0.6pre9 -> 0.6pre16

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

- Bump:
  - https://github.com/tildearrow/furnace/releases/tag/v0.6pre10
  - https://github.com/tildearrow/furnace/releases/tag/v0.6pre11
  - https://github.com/tildearrow/furnace/releases/tag/v0.6pre12
  - https://github.com/tildearrow/furnace/releases/tag/v0.6pre13
  - https://github.com/tildearrow/furnace/releases/tag/v0.6pre14
  - https://github.com/tildearrow/furnace/releases/tag/v0.6pre15
  - https://github.com/tildearrow/furnace/releases/tag/v0.6pre16
- Explicitly enable PortAudio audio output option, use system-installed one
- Explicitly enable SDL & OpenGL render options
  - Prefer GL or GLES based on hostPlatform

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/furnace/default.nix [hunk 1]
```diff
@@ -67,7 +75,12 @@ stdenv.mkDerivation rec {
     "-DSYSTEM_RTMIDI=ON"
     "-DSYSTEM_SDL2=ON"
     "-DSYSTEM_ZLIB=ON"
+    "-DSYSTEM_PORTAUDIO=ON"
     "-DWITH_JACK=${if withJACK then "ON" else "OFF"}"
+    "-DWITH_PORTAUDIO=ON"
+    "-DWITH_RENDER_SDL=ON"
+    "-DWITH_RENDER_OPENGL=ON"
```
Comment (wegank) [inline pkgs/applications/audio/furnace/default.nix:82]: Looks like this doesn't work on darwin?

File: pkgs/applications/audio/furnace/default.nix [hunk 2]
```diff
@@ -18,18 +18,22 @@
 , libjack2
 , withGUI ? true
 , Cocoa
+, portaudio
+, alsa-lib
+# Use GLES instead of GL
+, withGLES ? stdenv.hostPlatform.isAarch
```
Comment (wegank) [inline pkgs/applications/audio/furnace/default.nix:24]: Why aarch-only?

File: pkgs/applications/audio/furnace/default.nix [hunk 3]
```diff
@@ -18,18 +18,22 @@
...
-  version = "0.6pre9";
+  version = "0.6pre16";
 
   src = fetchFromGitHub {
     owner = "tildearrow";
     repo = "furnace";
     rev = "v${version}";
     fetchSubmodules = true;
-    sha256 = "sha256-i7/NN179Wyr1FqNlgryyFtishFr5EY1HI6BRQKby/6E=";
+    sha256 = "sha256-n66Bv8xB/0KMJYoMILxsaKoaX+E0OFGI3QGqhxKTFUQ=";
```
Comment (wegank) [inline pkgs/applications/audio/furnace/default.nix:36]:
```suggestion
    hash = "sha256-n66Bv8xB/0KMJYoMILxsaKoaX+E0OFGI3QGqhxKTFUQ=";
```

File: pkgs/applications/audio/furnace/default.nix (hunk 1 above)
Comment (OPNA2608) [reply pkgs/applications/audio/furnace/default.nix:82]: Will look into it tonight.

File: pkgs/applications/audio/furnace/default.nix (hunk 2 above)
Comment (OPNA2608) [reply pkgs/applications/audio/furnace/default.nix:24]: From personal experience with my own hardware, GLES support on ARM tends to be better than full GL support. Not sure what the situation is like on other drivers / SoCs / arches.

File: pkgs/applications/audio/furnace/default.nix (hunk 3 above)
Comment (OPNA2608) [reply pkgs/applications/audio/furnace/default.nix]: Oh true, applied.

File: pkgs/applications/audio/furnace/default.nix (hunk 1 above)
Comment (OPNA2608) [reply pkgs/applications/audio/furnace/default.nix:82]: From a quick look, might not be supported by upstream: https://github.com/tildearrow/furnace/blob/v0.6pre16/CMakeLists.txt#L70

**Pull request merged**